File: kmp-cli/src/main/resources/templates/kmp-template/kmptemplate/src/commonMain/kotlin/com/example/application/config/environment/EnvironmentConfig.kt
package com.example.kmptemplate.application.config.environment

/**
 * Contract for accessing environment-specific configuration information.
 */
internal interface EnvironmentConfig {
    /**
     * The base URL for API requests in this environment.
     */
    val baseUrl: String

    /**
     * The current environment (DEV, BETA, or PROD).
     */
    val environment: Environment
}

internal expect fun createEnvironmentConfig(environment: Environment): EnvironmentConfig


internal const val BETA_BASE_URL = ""
internal const val PROD_BASE_URL = ""

File: kmp-cli/src/main/resources/templates/kmp-template/apps/ios/KmpTemplate/KmpTemplate/Config/EnvironmentConfig.swift
import Foundation
import KmpTemplateKit

enum EnvironmentConfig {
    static var current: Environment {
        #if DEV
        return .dev
        #elseif BETA
        return .beta
        #elseif PROD
        return .prod
        #else
        return .dev
        #endif
    }
}
